pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -106,5 +127,61 @@ pub mod pallet {
 			)
 			.into())
 		}
+
+		// TODO(rodrigo): This extrinsic should be removed once the storage of destroyed contracts
+		// has been removed
+		#[pallet::call_index(1)]
+		#[pallet::weight({
+			let addresses_len = addresses.len() as u32;
+			<T as crate::Config>::WeightInfo::clear_suicided_storage(addresses_len, *limit)
+		})]
+		pub fn clear_suicided_storage(
+			origin: OriginFor<T>,
+			addresses: BoundedVec<H160, GetArrayLimit>,
+			limit: u32,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			ensure!(limit != 0, Error::<T>::LimitCannotBeZero);
+			ensure!(
+				addresses.len() != 0,
+				Error::<T>::AddressesLengthCannotBeZero
+			);
+
+			let mut limit = limit as usize;
+
+			for address in &addresses {
+				// Ensure that the contract is corrupted by checking
+				// that it has no code and at least one storage entry.
+				let suicided = pallet_evm::Suicided::<T>::contains_key(&address);
+				let has_code = pallet_evm::AccountCodes::<T>::contains_key(&address);
+				ensure!(
+					!suicided
+						&& !has_code && pallet_evm::AccountStorages::<T>::iter_key_prefix(&address)
+						.next()
+						.is_some(),
+					Error::<T>::ContractNotCorrupted
+				);
+
+				let deleted = pallet_evm::AccountStorages::<T>::drain_prefix(*address)
+					.take(limit)
```

Comment:
Maybe I'm missing some context, but I have some doubts regarding this limit:

- Why do we set a limit for the account storages to delete? Shouldn't we remove all the entries if the storage was already self-destructed? 
- What is the purpose of keeping only some storage entries and delete the others?

Reply:
1 - The total storage can be too big to fit in the maximum POV_SIZE allowed per block, this is why we do not do these migrations in a runtime upgrade, since it could possible stall the parachain.
2 - We are not keeping some storage entries, we expect to delete all storage entries from corrupted self-destructed contracts, we just remove the storage in a lazy way to avoid stalling the parachain. We can call the extrinsic multiple times until the account's storage has been deleted.

Reply:
Great thanks!